puppet/modules/jenkins_master/manifests/init.pp

```diff
@@ -191,6 +192,9 @@
     'zentimestamp'                       => {},
   }
 
+  package { $old_packages:
+    ensure => 'absent',
+  } ->
```

Comment:
This doesn't work. I've added a test to verify this. I'd also expect that you first install the new packages and then remove the old one so there's still a virtual that provides the JRE for Jenkins.